processor/spanmetricsprocessor/processor.go

```diff
@@ -282,8 +296,24 @@ func (p *processorImp) collectCallMetrics(ilm pdata.InstrumentationLibraryMetric
 		dpCalls.SetTimestamp(pdata.NewTimestampFromTime(time.Now()))
 		dpCalls.SetIntVal(p.callSum[key])
 
-		p.metricKeyToDimensions[key].CopyTo(dpCalls.Attributes())
+		if dimensions, okay := p.getDimensionsByMetricKey(key); okay {
+			dimensions.CopyTo(dpCalls.Attributes())
+		}
+	}
+}
+
+// getDimensionsByMetricKey get dimensions from `metricKeyToDimensions` cache
+func (p *processorImp) getDimensionsByMetricKey(k metricKey) (v *pdata.AttributeMap, okay bool) {
```

Comment:
Is it necessary to use named return params? It gets a little confusing with `okay` and `ok` vars mixed together in this func.